__tests__/app.test.js

```diff
@@ -5,32 +5,93 @@ const seed = require("../db/seeds/seed");
-                    expect(body.length).toBe(data.topicData.length)
-                });
+    });
+    test("Should return the correct data", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          body.forEach((topic) => {
+            expect(topic).toHaveProperty("slug", expect.any(String));
+            expect(topic).toHaveProperty("description", expect.any(String));
+          });
+        });
+    });
+  });
+});
+
+describe("GET /api", () => {
+  describe("Connecting to /api path", () => {
+    test("Should return status 200", () => {
+      return request(app).get("/api").expect(200);
+    });
+    test("Should return the data from the path", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body["GET /api"]).toMatchObject({
+            description:
+              "serves up a json representation of all the available endpoints of the api",
+          });
         });
-        test("Should return the correct data", () => {
-            return request(app).get("/api/topics").expect(200)
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(topic).toHaveProperty("slug", expect.any(String))
-                        expect(topic).toHaveProperty("description", expect.any(String))
-                    });
-                });
+    });
+    test("Should return the correct info for each endpoint (/api/topics)", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body["GET /api/topics"]).toMatchObject({
+            "description": "serves an array of all topics",
+            "queries": [],
+            "exampleResponse": {
+              "topics": [{ "slug": "football", "description": "Footie!" }]
+            }
+          });
         });
     });
-});
\ No newline at end of file
+    test("Should return the correct info for each endpoint (/api/articles)", () => {
+        return request(app)
+          .get("/api")
+          .expect(200)
+          .then(({ body }) => {
+            expect(body["GET /api/articles"]).toMatchObject({
+                "description": "serves an array of all articles",
+                "queries": ["author", "topic", "sort_by", "order"],
+                "exampleResponse": {
+                  "articles": [
+                    {
+                      "title": "Seafood substitutions are increasing",
+                      "topic": "cooking",
+                      "author": "weegembump",
+                      "body": "Text from the article..",
+                      "created_at": "2018-05-30T15:59:13.341Z",
+                      "votes": 0,
+                      "comment_count": 6
+                    }
+                  ]
+                }
+              });
+          });
+      });
+  });
+});
```

Comment:
This way of testing isn't wrong but it's going to require you to write a new test every time you add a new endpoint. Can you think of a way you could dynamically test this?